src/provisioning/api.md

```diff
@@ -0,0 +1,290 @@
+        "avatarUrl": "https://avatars.githubusercontent.com/u/8418310?v=4"
+    ]
+}
+```
+
+### GET /v1/github/orgs/{orgName}/repositories?userId={userId}&page={page}&perPage={perPage}
+
+Request a list of all repositories a user is a member of in the given org. The `owner` and `name` value of a repository can be given to create a new GitHub connection.
+
+This request is paginated, and `page` sets the page (defaults to `1`) while `perPage` (defaults to `10`) sets the number of entries per page.
+
+This request can be retried until the number of entries is less than the value of `perPage`.
+
+### Response
+
+```json5
+{
+    "loggedIn": true,
+    "repositories": [
+        "name": "matrix-hookshot",
+        "owner": "half-shot",
+        "fullName": "half-shot/matrix-hookshot",
+        "avatarUrl": "https://avatars.githubusercontent.com/u/8418310?v=4",
+        "description": "A bridge between Matrix and multiple project management services, such as GitHub, GitLab and JIRA. "
+    ]
+}
+```
+
+### GET /v1/github/repositories?userId={userId}&page={page}&perPage={perPage}
+
+Request a list of all repositories a user is a member of (including those not belonging to an org). The `owner` and `name` value of a repository can be given to create a new GitHub connection.
+
+This request is paginated, and `page` sets the page (defaults to `1`) while `perPage` (defaults to `10`) sets the number of entries per page.
+
+This request can be retried until the number of entries is less than the value of `perPage`.
+
+### Response
+
+```json5
+{
+    "loggedIn": true,
+    "repositories": [
+        "name": "matrix-hookshot",
+        "owner": "half-shot",
+        "fullName": "half-shot/matrix-hookshot",
+        "avatarUrl": "https://avatars.githubusercontent.com/u/8418310?v=4",
+        "description": "A bridge between Matrix and multiple project management services, such as GitHub, GitLab and JIRA. "
+    ]
+}
+```
+
+## JIRA
+
+
+### GET /v1/jira/oauth?userId={userId}
+
+
+Request an OAuth url for the given user. Once the user has completed the steps in the OAuth process,
+the bridge will be granted access.
+
+### Response
+
+```json5
+{
+    "url": "https://auth.atlassian.com/authorize?..."
+}
+```
+
+### GET /v1/jira/account?userId={userId}
+
+
+Request the status of the users account. This will return a `loggedIn` value to determine if the
+bridge has a JIRA identity stored for the user, and any instances they have access to. Note that if a 
+user does not have access to an instance, they can authenticate again to gain access to it (if they are able
+to consent).
+### Response
+
+```json5
+{
+    "loggedIn": true,
+    "instances": [
+        "name": "acme",
+        "url": "https://acme.atlassian.net"
+    ]
+}
+```
+
+### GET /v1/jira/instances/{instanceName}/projects?userId={userId}
+
+Request a list of all projects a user can see in a given instance. The `url` value of a project can be given to create
+a new JIRA connection.
+### Response
```

Comment:
This actually just returns an unwrapped array, as demonstrated by this helper function:

```
    public async getProjects(instanceName: string): Promise<HookshotJiraProject[]> {
        const bridge = await this.getDefaultBridge();
        return this.doProvisionRequest<HookshotJiraProject[]>(bridge, "GET", `/v1/jira/instances/${instanceName}/projects`);
    }
```